### PR TITLE
Add datasets.d mode

### PR DESCRIPTION
**Pull request by srstsavage**
Add a datasets.d mode which composes a datasets.xml
file from constituent dataset elements contained in
the /datasets.d directory and environment variables
prefixed with ERDDAP_DATASETS_. See README.md for
more details.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -25,6 +25,38 @@ if [ "$1" = 'start-tomcat.sh' ] || [ "$1" = 'catalina.sh' ]; then
...
+      #set top level datasets.xml config with ERDDAP_DATASETS_* env vars
+      env | grep -oP '(?<=^ERDDAP_DATASETS_).*' | while read -r e; do
+        k=$(echo "$e" | cut -d= -f1);
+        v=$(echo "$e" | cut -d= -f2-);
+        xmlstarlet edit --inplace --subnode /erddapDatasets --type elem --name "$k" --value "$v" "${DATASETS_XML}"
+      done
+
+      #empty edit for formatting
+      xmlstarlet edit --inplace "${DATASETS_XML}"
+    fi
```
Comment (abkfenris) [inline entrypoint.sh]: How about extracting the `datasets.xml` creation into a standalone script? That way `datasets.d` mode can also be used for testing and in-place updates without restarting the container?

Comment (srstsavage) [reply entrypoint.sh]: Yes, great idea. I pushed a commit making this change (extracting datasets.d logic to `datasets.d.sh`).

**Comment by MathewBiddle** [comment]
If it's of interest, I wrote a python function to do a similar activity:
https://github.com/MathewBiddle/sandbox/blob/main/xml_insert.py
which I run using 
https://github.com/MathewBiddle/sandbox/blob/main/script2insert.py

**Comment by srstsavage** [comment]
Thanks Matt, nice to see another implementation (and evidence that others want this functionality!). I considered using python to wrangle the XML here as it would be straightforward/familiar/powerful, but I wanted to keep the dependency introduction as small as possible (`xmlstarlet` has a much smaller footprint than a python installation). Granted that python3 is currently available in this image, but we might consider shrinking it in the future...

**Review by kwilcox — APPROVED** [review]

**Pull request merged**